src/solvers/refinement/string_refinement.cpp

```diff
@@ -1855,28 +1854,12 @@ find_index(const exprt &expr, const exprt &str, const symbol_exprt &qvar)
     return false;
   };
 
-  auto it = std::find_if(
-    expr.depth_begin(), expr.depth_end(), index_str_containing_qvar);
-  if(it == expr.depth_end())
-    return {};
-  const exprt &index = to_index_expr(*it).index();
-
-  // Check that there are no two such indexes
-  it.next_sibling_or_parent();
-  while(it != expr.depth_end())
+  for(auto it = expr.depth_begin(); it != expr.depth_end(); ++it)
```

Comment:
This could be `for_each(expr.depth_begin(), ...` as `it` is not used as iterator.